be/src/vec/exec/vaggregation_node.cpp

```diff
@@ -208,20 +229,36 @@ void AggregationNode::_init_hash_method(std::vector<VExprContext*>& probe_exprs)
         if (use_fixed_key) {
             if (has_null) {
                 if (std::tuple_size<KeysNullMap<UInt64>>::value + key_byte_size <= sizeof(UInt64)) {
-                    _agg_data.init(AggregatedDataVariants::Type::int64_keys, has_null);
+                    if (_is_streaming_preagg)
+                        _agg_data.init(AggregatedDataVariants::Type::int64_keys, has_null);
+                    else
+                        _agg_data.init(AggregatedDataVariants::Type::int64_keys_phase2, has_null);
```

Comment:
if _is_streaming_preagg == false, this  means current AggNode maybe the second phase of a streaming agg, but also may be a first phase of BlockAgg.

Reply:
Yes, here replaced this with `_is_merge` to check.